File: milestone 2/series4/brownp.m
function x = brownp(T,N,M)
% x = brownp(T,N)
% x = brownp(T,N,M)
% find M paths of Brownian motion, default M=1
% each column of x contains values at t=T/N, 2*T/N, ... , T
if nargin==2
  M=1;
end
x = [zeros(1,M);sqrt(T/N)*cumsum(randn(N,M))];

File: milestone 2/series4/brownp.py
import numpy as np

def brownp(T, N, M=1):
    """
    Find M paths of Brownian motion, default M=1.
    Each column of x contains values at t=T/N, 2*T/N, ... , T.
    """
    x = np.zeros((N+1, M))
    x[1:] = np.sqrt(T/N) * np.cumsum(np.random.randn(N, M), axis=0)
    return x